### DISCUSSION TITLE
MAUI port roadmap

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [xamarin/XamarinCommunityToolkit#1518] [Q&A] [answered]
With Preview 7 (in a month from now) MAUI [should be feature complete](https://github.com/dotnet/maui/wiki/Roadmap#net-maui-in-net-6-preview-7-august-2021), so people might start migrating their apps in the near future.

Are there any plans or roadmap regarding poring this toolkit to MAUI?

**Comment by pictos** [comment, marked as answer]
Yes, it's. And they are amazing❣ You can see our plans [here](https://devblogs.microsoft.com/xamarin/the-future-of-xamarin-community-toolkit/).

**Reply by maxkoshevoi** [reply]
Great news! Missed that article. 

Are there any guidelines for contributors to https://github.com/CommunityToolkit/Maui? Like what's left to port, how platform specific code should be organized, etc. I'd like to help with migration, but not sure what exactly to help with.

Also what's going to happen with currently open enhancement PRs if they are still open in September?
Are there any plans to go through open PRs and review / merge them if everything is ok? I have three PRs opened that didn't see any activity for one to three month now. I'd be happy to work with someone and resolve any comments so they can be merged.

**Reply by pictos** [reply]
@maxkoshevoi AFAIK we don't have any guideline or roadmap yet. We also need to see what makes sense to port, since some stuff that we have here will be supported out of the box on MAUI (like shadows).

**Reply by maxkoshevoi** [reply]
@pictos Got it, waiting until there are some issues open regarding what needs to be ported then.

Are there any plans regarding currently open PRs (especially ones without unresolved comments)?

**Reply by jfversluis** [reply]
We don't have any specific plans for the open PRs right now. My personal feeling is that everything we merge now is a risk of destabilizing what is there and will also cause us duplicate work by merging it now and therefore require ourselves to also port it to the .NET MAUI version.

I think it's also mentioned in the blog post that Pedro linked that we will stop adding new features as of later this month and then follow the same path as Forms to just release bugfixes and stability fixes. 

I appreciate the enthusiasm, we're trying to set things up for the new Community Toolkit while maintaining two others so please have some patience :)

**Reply by maxkoshevoi** [reply]
@jfversluis Got it, so feature PRs are low priority and unlikely to be merged/looked at.

What about bugfixes? 
- https://github.com/xamarin/XamarinCommunityToolkit/pull/1357
- https://github.com/xamarin/XamarinCommunityToolkit/pull/1223

> I appreciate the enthusiasm, we're trying to set things up for the new Community Toolkit while maintaining two others

If I understand you correctly, they are also postponed until two new toolkits are ready, and will be merged here first and then ported into them. Is my understanding correct?

**Reply by GUOLDEV** [reply]
@maxkoshevoi I was about to ask about the roadmap and found your discussion 👍 @jfversluis @pictos just a suggestion, what about adding some comments to the XCT description page about not adding new features and focus on the bug fixes? The idea is getting the community members efforts and focus on the bug fixes and stability so we have a smooth transition to MAUI. Thoughts?

**Reply by GUOLDEV** [reply]
I was thinking about a bug-bash, so we can get bugs fixed as much as we can, I have checked old bugs (issues report) and some of them maybe are already fixed or hard to reproduce, so we can focus on the ones that we can reproduce or just waiting review.

**Reply by pictos** [reply]
Hey, @GUOLDEV this is what we have planned, just need to double-check what's the deadline to accept new features. After this deadline, we will focus just on bug fixes. In terms of `feature requests`, we'll see what makes sense to move to the MauiCommunityToolkit (MCT).  

Sorry if we passed a feeling that we don't have anything defined, the truth is that we've been working in backgrounds to create the base of the MCT and start the migration. In September we will share more stuff with contributors.
